### PR TITLE
SEARCH-1298 Allows to use ref on the connected Autosuggest component

### PR DESCRIPTION
**Pull request by vrocha-daitan**
## Description
Fixes an issue where the info barrier modal is opened on top of the header search dropdown and then it becomes impossible to close the dropdown after clicking on the modal's "Close" button. See gifs for more clarity.
[SEARCH-1298](https://perzoinc.atlassian.net/browse/SEARCH-1298)

## Earlier
1298-before

## Now
1298-after

## Approach
[comment]: # (How does this change address the problem?)
#### Problem with the code:
- After closing a modal, focus doesn't return to the element that was previously focused -- in this case, the header search dropdown.
- In `symphony-react-autosuggest`, there's a flag called `justClickedOnSuggestion` which controls whether or not the dropdown should be closed. This flag should be reset to `false` when clicking on the search dropdown, but that never happens in the bug scenario (room prompt + info barrier modal). 

#### Fix:
- Forcefully set the above-mentioned flag to `false` when the infobarrier is closed.
- Forcefully return focus to the header search when the infobarrier is closed.

## Related PRs
[comment]: # (List the related PRs against other branches.)

Repo | PR #
------ | ------ 
SFE-Client-App | https://github.com/SymphonyOSF/SFE-Client-App/pull/13517

### CODE REVIEW

**Comment by vrocha-daitan** [comment]
@lsouza-daitan @portiz-daitan @NitinSingh2020 @VikasShashidhar @keerthi16 @adarshvadwitiya please review.

**Review by keerthi16 — APPROVED** [review]
LGTM 👍

**Review by adarshvadwitiya — APPROVED** [review]
LGTM

File: src/AutosuggestContainer.js
Comment (pamela-encora) [inline src/AutosuggestContainer.js:74]: Please set a default value for this prop

**Review by VikasShashidhar — APPROVED** [review]

File: src/AutosuggestContainer.js
Comment (vrocha-daitan) [reply src/AutosuggestContainer.js:74]: Done, thanks.

**Review by pamela-encora — APPROVED** [review]
LGTM

**Comment by vrocha-daitan** [comment]
@NitinSingh2020 @satheeshkr Could you please help merge this if it looks good? This is an essential part of the fix for https://perzoinc.atlassian.net/browse/SEARCH-1298 and I don't have write access to this repo.

**Comment by NitinSingh2020** [comment]
Published `symphony-react-autosuggest@3.7.3-symphony.10`

**Pull request merged**
